api/desecapi/views.py

```diff
@@ -127,52 +130,37 @@ class RRsetDetail(generics.RetrieveUpdateDestroyAPIView):
     serializer_class = RRsetSerializer
     permission_classes = (permissions.IsAuthenticated, IsDomainOwner,)
 
-    def get_queryset(self):
-        name = self.kwargs['name']
-        subname = self.kwargs['subname']
-        type = self.kwargs['type']
-
-        return RRset.objects.filter(domain__name=name, subname=subname, type=type)
-
     def delete(self, request, *args, **kwargs):
         try:
-            return super().delete(request, *args, **kwargs)
-        except django.core.exceptions.ValidationError as e:
-            ex = ValidationError(detail={"detail": str(e)})
-            ex.status_code = status.HTTP_409_CONFLICT
-            raise ex
+            super().delete(request, *args, **kwargs)
+        except Http404:
+            pass
+        return Response(status=status.HTTP_204_NO_CONTENT)
 
-    def get(self, request, *args, **kwargs):
+    def get_queryset(self):
+        name = self.kwargs['name']
         subname = self.kwargs['subname']
         type = self.kwargs['type'].upper()
 
-        # TODO Do we need this functionality at all? We could just disregard the SOA record completely.
-        if type == 'SOA' and subname == '':
-            name = self.kwargs['name']
+        if type == 'SOA':
+            raise PermissionDenied("You cannot tamper with the SOA record.")
```

Comment:
Should we implement this as a Django permission? The `DoNotQuerySOARecordPermission` or something?